cfme/tests/configure/test_default_views_infra.py

```diff
@@ -50,6 +50,20 @@ def set_and_test_default_view(group_name, view, page):
     assert tb.is_active(view), "{} view setting failed".format(view)
     DefaultView.set_default_view(group_name, old_default)
 
+
+def check_vm_visibility():
+    view = navigate_to(Vm, 'Explore')
+    value = view.sidebar.vmstemplates.tree.read_contents()
+    # Selecting the path and last Vm in the accordion to perform click
+    tree_root = value[0]
+    tree_provider = value[1][0][0]
+    tree_datacenter = value[1][0][1][0][0]
+    length = len(value[1][0][1][0][1])
+    tree_vm = value[1][0][1][0][1][length - 1]
+    view.sidebar.vmstemplates.tree.click_path(tree_root,
+                                              tree_provider, tree_datacenter, tree_vm)
+    return "Instance" in view.title.text
```

Comment:
what exactly you check in this function ?

Reply:
I am checking if we click on the VM in the explorer tree we should land on different page i.e. Details page which include 'Instance' in it's title.

Here I am reading the Vms in explorer tree and finally selecting the last VM and checking that are we landing to the Vms details page or not. If the new page is loaded then with help of the title I am able to verify if it is the required page or not.